File: GC.Domain/Contacts/GardenContacts/GardenContact.cs
using GC.Domain.Gardens;
using System;

namespace GC.Domain.Contacts.GardenContacts
{
    public class GardenContact
    {
        public Guid Id { get; }
        public Gardener Gardener { get; }
        public GardenContactType Type { get; }
        public String PhoneNumber { get; }

        public GardenContact(Guid id, Gardener gardener, GardenContactType type, string phoneNumber)
        {
            Id = id;
            Gardener = gardener;
            Type = type;
            PhoneNumber = phoneNumber;
        }
    }
}

File: GC.WebSpace/ClientApp/domain/contacts/gardenContacts/gardenContact.ts
import { Gardener } from "../../gardens/gardeners/gardener";
import { GardenContactType } from "./gardenContactType";

export class GardenContact {
    constructor(
        public readonly id: string,
        public readonly gardener: Gardener,
        public readonly type: GardenContactType,
        public readonly phoneNumber: string,
    ) { }
}

export namespace GardenContact {
    export function toModel(object: any): GardenContact {
        return new GardenContact(object.id, Gardener.toModel(object.gardener), object.type, object.phoneNumber);
    }

    export function toModels(objects: any[]): GardenContact[] {
        return objects.map(toModel);
    }
}